### PR TITLE
Add sentry capture exception to resolver methods

### PR DESCRIPTION
**Pull request by codingone21**
Closes: https://github.com/onflow/next-docs-v1/issues/461

## Description
Enabling sentry reporting, which is currently all ignored due to Middleware. Unfortunately, logrus doesn't have an updated package for sentry integration, so the easiest solution seems to be notifying Sentry at the highest level of error (resolver methods)

Alternative solution:
We could try integrating sentry-go and logrus middleware using something like this: https://github.com/onrik/logrus. I am currently trying to make this work locally.
______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by psiemens — COMMENTED** [review]
> which is currently all ignored due to Middleware

Hmm, do you know why the middleware is blocking the error reporting? Is it a known issue?

File: compute/errors.go [hunk 1]
```diff
@@ -29,6 +30,7 @@ import (
 )
 
 func ExtractProgramErrors(err error) (result []model.ProgramError) {
+	sentry.CaptureException(err)
```
Comment (sideninja) [inline compute/errors.go:33]: I'm not sure these are errors we want to catch in Sentry, they are related to the program parsing and execution.

**Comment by sideninja** [comment]
Hey @codingone21 wouldn't adding sentry to chi router as middleware be best? This is written in docs https://docs.sentry.io/platforms/go/guides/http/#install 

So doing something like:
```go
	err := sentry.Init(sentry.ClientOptions{
		Dsn:   "", // Set DSN here or set SENTRY_DSN environment variable
		Debug: true,
	})
	if err != nil {
		log.Fatalf("sentry.Init: %s", err)
	}
	defer sentry.Flush(time.Second)

	sentryMiddleware := sentryhttp.New(sentryhttp.Options{
		Repanic: true,
	})

	r := chi.NewRouter()
	r.Use(sentryMiddleware.Handle)
	
	// rest of code
```

Is there a problem with this approach?

**Comment by sideninja** [comment]
Did you manage to validate this working? Ping me when the Sentry reports are in.

**Comment by codingone21** [comment]
> Did you manage to validate this working? Ping me when the Sentry reports are in.

I got a sentry report when I panic in handler: https://sentry.io/organizations/dapperlabs/issues/3443167930/?project=6398442&query=is%3Aunresolved&statsPeriod=14d

File: middleware/errors/errors.go [hunk 2]
```diff
@@ -51,6 +52,8 @@ func Middleware(entry *logrus.Entry) graphql.RequestMiddleware {
 			} else if err != nil {
 				contextEntry.WithError(err).Warnf("GQL Request Client Error: %v err = %+v", err.Extensions["general_error"], err)
 			}
+
+			localHub.RecoverWithContext(ctx, err)
```
Comment (codingone21) [inline middleware/errors/errors.go:56]: This catches GQL errors like this: https://sentry.io/organizations/dapperlabs/issues/3453822061/?project=6398442&query=&statsPeriod=14d

File: resolver.go [hunk 3]
```diff
@@ -61,6 +62,9 @@ func NewResolver(
 	computer *compute.Computer,
 	auth *auth.Authenticator,
 ) *Resolver {
+	defer sentry.Flush(2 * time.Second)
+	defer sentry.Recover()
```
Comment (codingone21) [inline resolver.go:66]: This is for handling panics in the main routine like this
https://sentry.io/organizations/dapperlabs/issues/3453939358/?project=6398442&query=&statsPeriod=14d

File: server/server.go [hunk 4]
```diff
@@ -90,16 +92,18 @@ func main() {
...
+		AttachStacktrace: sentryConf.AttachStacktrace,
 	})
 
 	if err != nil {
 		log.Fatalf("sentry.Init: %s", err)
 	}
 
 	defer sentry.Flush(2 * time.Second)
-	
+	defer sentry.Recover()
```
Comment (codingone21) [inline server/server.go:105]: This is for handling panics in the main routine: 
https://sentry.io/organizations/dapperlabs/issues/3453939358/?project=6398442&query=&statsPeriod=14d

File: server/server.go [hunk 5]
```diff
@@ -174,14 +178,29 @@ func main() {
 			cookieStore.Options.SameSite = http.SameSiteNoneMode
 		}
 
+		// Create a new hub for this subroutine and bind current client and handle to scope
+		localHub := sentry.CurrentHub().Clone()
```
Comment (codingone21) [inline server/server.go:182]: This is for handling panics in the /query subroutine

**Comment by codingone21** [comment]
> Did you manage to validate this working? Ping me when the Sentry reports are in.

We can go over them together in slack, here are some errors I triggered on sentry: https://sentry.io/organizations/dapperlabs/issues/?project=6398442&query=&statsPeriod=14d

File: server/server.go (hunk 4 above)
Comment (sideninja) [inline server/server.go:105]: 👍

**Review by sideninja — APPROVED** [review]

**Pull request merged**